nmrfx-analyst-gui/src/main/java/org/nmrfx/analyst/gui/AnalystApp.java

```diff
@@ -138,7 +134,7 @@ public void start(Stage stage) throws Exception {
         MainApp.setAnalyst();
         mainApp = this;
         analystApp = this;
-        FXMLController controller = FXMLController.create(stage);
+        FXMLController.create(stage).addCanvasKeyEventHandler(new CopyMoleculeEventHandler());
```

Comment:
I'm not sure if this is the best setup, the CopyMoleculeEventHandler has to be added to the FXMLController each time a new one is created, but couldn't put this code in the `nmrfx-processor-gui` module because the canvas molecule code is in the `nmrfx-analyst-gui` module (would be circular dependencies), Should we move some of the molecule code to `nmrfx-processor-gui`? Or leave it where it is and set the event handler like this?, ...or other suggestions?

Reply:
At some point I'd like to reevaluate the whole separation into nmrfx-processor-gui and nmrfx-analyat-gui.  There was a rational early on, where there were really two programs, NMRFx Processor that did just processing, and NMRFx Analyst that added lots of other features for analysis.  But we've pretty much gotten rid of NMRFx Processor" as a standalone GUI program.  But, I think we should make the addCanvasKey have a more generic function (than just molecules) and possibly register DataFormats from analyst.  So we'd register a DataFormat type for molecules from analyst code.  Needs a little more thought to see if this works.  It would also break your parsing of text as molecule.  But for that we may need a parser that can decide what the text represents (could be a molecule as done here, but could be other things like the text description of a spectrum).

Reply:
Created a more general CanvasKeyEventHandler which can call different DataFormat handlers. I made a separate plain text handler so the pasting of plain text molecules still works, and we can add more discerning parsing there in the future if we need to paste something else in plain text.

Reply:
There is an existing KeyHandler class in org.nmrfx.processor.gui.spectra.  Seems like we should handle the Paste events there rather than adding a new CanvasKeyEventHandler.
Also, instead of calling addCanvasDataFormatHandler on each created instance of FXMLController couldn't we just make that a static method add to static map in FXMLController?